src/main/java/org/kitteh/irc/client/library/implementation/IRCClient.java

```diff
@@ -103,9 +105,20 @@ protected void processElement(@Nonnull String element) {
     private final ActorProvider actorProvider = new ActorProvider(this);
 
     private Map<Character, ModeStatus<UserMode>> userModes;
+    private StsMachine stsMachine;
+    private boolean usingSts = false;
 
     IRCClient(@Nonnull Config config) {
         this.config = config;
+        if (this.config.get(Config.STS_STORAGE_MANAGER) != null) {
+            this.configureSts();
+        } else if (!this.config.getNotNull(Config.SSL)) {
+            throw new KittehAuthorNagException(
```

Comment:
You're also asking them to consider it, but not really leaving a choice in the matter.